lib/Core/Executor.cpp

```diff
@@ -1680,6 +1680,37 @@ void Executor::executeCall(ExecutionState &state, KInstruction *ki, Function *f,
       break;
     }
 
+    case Intrinsic::fma:
+    case Intrinsic::fmuladd: {
+      // only fma supports vectors!
+      if (isa<VectorType>(i->getOperand(0)->getType()))
+        return terminateStateOnExecError(
+            state, "llvm.fma with vectors is not supported");
+
+      ref<ConstantExpr> op1 =
+          toConstant(state, eval(ki, 1, state).value, "floating point");
+      ref<ConstantExpr> op2 =
+          toConstant(state, eval(ki, 2, state).value, "floating point");
+      ref<ConstantExpr> op3 =
+          toConstant(state, eval(ki, 3, state).value, "floating point");
+
+      if (!fpWidthToSemantics(op1->getWidth()) ||
+          !fpWidthToSemantics(op2->getWidth()) ||
+          !fpWidthToSemantics(op3->getWidth()))
+        return terminateStateOnExecError(state,
+                                         "Unsupported fmuladd operation");
```

Comment:
Can you add a more informative message, e.g., adding that only IEEE single and double precision is supported?

Reply:
And should this be "Unsupported fma or fmuladd operation"?

Reply:
What do you mean by the first comment? Both `fma` and `fmuladd` support `float`, `double` and `x86_fp80`. The documentation of LLVM does not mention `fmuladd` supporting `x86_fp80` but a following experiment shows, that it it, in fact, does:
```c
int main(void) {
    long double a = 4, b = 4, c = 2;
    return a * b + c;
}
```
```llvm
$ clang fma.c -emit-llvm -S -O0 -o - | grep fmuladd
  %8 = call x86_fp80 @llvm.fmuladd.f80(x86_fp80 %5, x86_fp80 %6, x86_fp80 %7)
declare x86_fp80 @llvm.fmuladd.f80(x86_fp80, x86_fp80, x86_fp80) #1
```

That said, I will add a comment mentioning this discrepancy in LLVM's documentation and my apologies for the confusion.

Reply:
@lzaoral Thanks -- my main point is that it would be useful to tell the user which kind of fma/fmuladd is unsupported here.

Reply:
Got it, addressed in 3567557.